src/main/kotlin/org/opensearch/commons/alerting/model/BaseAlert.kt

```diff
@@ -138,7 +137,7 @@ open class BaseAlert(
                         }
                     }
                     START_TIME_FIELD -> startTime = requireNotNull(xcp.instant())
-                    END_TIME_FIELD -> endTime = xcp.instant()
+                    END_TIME_FIELD -> endTime = requireNotNull(xcp.instant())
```

Comment:
yes, we need it and it will always be there because it's calculated based on the startTime plus the duration of the time window for the correlation rule. So, it's a little different from what we have for Alerting. This startTime and endTime window will help us preventing from creating multiple alerts in that timeframe.